src/main/java/View/playlistView.java

```diff
@@ -55,40 +52,32 @@ public playlistView(User user){
         // whatever song is playing
         // TODO
 
-
+        // set screen visible
         this.initializeFrame();
-    }
 
+    }
 
     /**
      * Invoked when an action occurs.
      * @param e the event to be processed
      */
     @Override
     public void actionPerformed(ActionEvent e) {
-
         if (e.getSource() == this.spaceButton){
-
-            if (!this.spacePlaying) {                           // clicking on button when space is playing does nothing
-
+            if (!this.spacePlaying){   // clicking on button when space is playing does nothing
                 String message = this.musicEngineControllerViewModel.callPlaySpace();
                 this.spaceButton.setText(message);
-
             }
-        } else if (e.getSource() == this.searchButton) {
+        } else if(e.getSource() == this.searchButton){
             String searchText = this.searchBar.getText();
             this.searchViewModel.search(searchText);
-
-            this.jframe.dispose();
-            new searchOutputView(searchText);
-
-        } // TODO -- NOTE: add your action commands as an else-if to this if statement
-
+            System.out.println(searchText);
+            new searchOutputView(searchText, this);
```

Comment:
same comment as what I wrote for searchOutputView. Passing in the playlistView instance is not needed for searchOutput view and will avoid conflicts when merging raf's pull request as well.